spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java

```diff
@@ -1183,6 +1226,17 @@ void customizeUndertow(ServerProperties serverProperties,
 			if (serverProperties.getMaxHttpPostSize() > 0) {
 				customizeMaxHttpPostSize(factory, serverProperties.getMaxHttpPostSize());
 			}
+
+			customizeConnectionTimeout(serverProperties, factory);
+		}
+		private void customizeConnectionTimeout(final ServerProperties serverProperties,
+													final UndertowEmbeddedServletContainerFactory factory) {
+			factory.addBuilderCustomizers(new UndertowBuilderCustomizer() {
+				@Override
+				public void customize(Builder builder) {
+					builder.setSocketOption(UndertowOptions.REQUEST_PARSE_TIMEOUT, serverProperties.getConnectionTimeout());
```

Comment:
`REQUEST_PARSE_TIMEOUT` is described as "The maximum allowed time of reading HTTP request in milliseconds.". Should this not be `NO_REQUEST_TIMEOUT` which is described as "The amount of time the connection can be idle with no current requests before it is closed"